algebird-test/src/test/scala/com/twitter/algebird/CountMinSketchTest.scala

```diff
@@ -272,6 +283,46 @@ abstract class CMSTest[K: Ordering: CMSHasher: Numeric] extends WordSpec with Ma
 
   "A Top-% Count-Min sketch implementing CMSHeavyHitters" should {
 
+    "create correct sketches out of a single item" in {
+      forAll{ (x: Int) =>
+        val data = x.toK[K]
+        val cmsMonoid = {
+          val anyHeavyHittersPct = 0.1 // exact setting not relevant for this test
+          TopPctCMS.monoid[K](EPS, DELTA, SEED, anyHeavyHittersPct)
+        }
+        val topCms = cmsMonoid.create(data)
+        topCms.totalCount should be(1)
+        topCms.cms.totalCount should be(1)
+        topCms.frequency(x.toK[K]).estimate should be(1)
+        // Poor man's way to come up with an item that is not x and that is very unlikely to hash to the same slot.
+        val otherItem = x + 1
+        topCms.frequency(otherItem.toK[K]).estimate should be(0)
+        // The following assert indirectly verifies whether the counting table is not all-zero (cf. GH-393).
+        topCms.innerProduct(topCms).estimate should be(1)
+      }
+    }
+
+    "create correct sketches out of a single-item stream" in {
+      forAll{ (x: Int, y: Int) =>
```

Comment:
You were right, this shadowing should not have been there.  The outer x/y loop was a relic from the previous test code.  Dumb copy-paste mistake of mine!